packages/data-stores/src/plans/types.ts

```diff
@@ -91,6 +91,8 @@ export interface PlanPricing {
 		monthly: number | null;
 		full: number | null;
 	};
+
+	discountReasonCode?: string;
```

Comment:
Similar - if this is only relevant to `/sites/[site]/plans` endpoint, then let's maybe move it below under `SitePlanPricing`